### ISSUE TITLE
Get authenticated user into plugin custom route

### ISSUE THREAD

**Issue by geco** [Chocobozzz/PeerTube#4174]
Hello @Chocobozzz , there is a way to get the current peertube authenticated user into plugin custom route?
I'm doing as in the code snippet below, but user is undefined, also i've seen res.locals and req.session are empty.

Server side (main.js):

async function register ({
  getRouter,
  peertubeHelpers
}) {

  const router = getRouter()

  router.get('/myroute', async (req, res) => {
    const user = await peertubeHelpers.user.getAuthUser(res); // THIS IS UNDEFINED
  
  });

}

**Comment by Chocobozzz** [comment]
Hello,

What is your peertube version?

**Comment by geco** [comment]
> Hello,
> 
> What is your peertube version?

3.2.1

**Comment by Chocobozzz** [comment]
Then your code snippet should work :thinking: Check your user is correctly authenticated in the client.

**Comment by geco** [comment]
> Then your code snippet should work 🤔 Check your user is correctly authenticated in the client.

Yes, the client is authenticated. Perhaps the problem is that from the plugin client side I call plugin backend route without Authorization Bearer (using XMLHttpRequest). But this will be true for every plugin client side? How can I authenticate my plugin on the client and use Authorization Bearer?

**Comment by geco** [comment]
I've found it: https://docs.joinpeertube.org/contribute-plugins?id=auth-header
I have to authenticate the client call
Thank you very much @Chocobozzz
